python/cudf_polars/cudf_polars/callback.py

```diff
@@ -217,14 +217,16 @@ def validate_config_options(config: dict) -> None:
         If the configuration contains unsupported options.
     """
     if unsupported := (
-        config.keys() - {"raise_on_fail", "parquet_options", "executor"}
+        config.keys()
+        - {"raise_on_fail", "parquet_options", "parallel_options", "executor"}
     ):
         raise ValueError(
             f"Engine configuration contains unsupported settings: {unsupported}"
         )
     assert {"chunked", "chunk_read_limit", "pass_read_limit"}.issuperset(
         config.get("parquet_options", {})
     )
+    assert {"num_rows_threshold"}.issuperset(config.get("parallel_options", {}))
```

Comment:
I'd like to nest all multi-gpu options within the `"parallel_options"` moving forward (to avoid adding more top-level keys).

Reply:
We might imagine that these options are executor-specific, does it make sense to have a nesting that is:

```
executor: str | tuple[str, dict]
```
So the executor argument is either a name, or a `("name", name-specific-options)`?

Reply:
That seems fine to me. Any opinion on this @pentschev ?

I do think it's a good idea to consider how the number of these options will inevitably grow over time (and that they will **probably**  be executor-specific).

Reply:
Hmm. The `str | tuple[str, dict]` logic actually feels a bit clumsy when I think about how to implement it.

How about we just rename `"parallel_options"` to `"executor_options"` (to make it clear that the options are executor-specific)? This still allows us to validate that the specified arguments are actually supported by the "active" executor.

Reply:
As much as I agree that it is indeed clumsy it feels like we'll soon need to have nested options and inevitably make `"executor_options"` require accepting `str | tuple[str, dict]`, so we may as well just do that in `executor` and with that allow as many levels of nested options as needed as part of `executor`. I think a better alternative may be an abstract base class `Executor` that we can specialize with the options we need for each executor.

Reply:
>I think a better alternative may be an abstract base class Executor that we can specialize with the options we need for each executor.

I do think this is the best long-term solution, but I also don't think it will be difficult to migrate from the `"executor_options"` approach currently used in this PR. 

I don't think I understand why it is inevitable that `"executor_options"` would need to accept `str | tuple[str, dict]`. However, I **do** see why it would be useful to attach all executor-specific options to an `Executor` object. That said, I don't really want to deal with serialization/etc in this PR :)

Reply:
> I don't think I understand why it is inevitable that `"executor_options"` would need to accept `str | tuple[str, dict]`.

It's possible I'm overestimating the amount of options we'll end up introducing here, but once we need nested options we'll need something more complex like the `tuple[str, dict]`, or the abstract base class. Thus why I think it's inevitable.